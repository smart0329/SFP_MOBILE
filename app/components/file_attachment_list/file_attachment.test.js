// Copyright (c) 2019-present Smart.Inc. All Rights Reserved.
// See LICENSE.txt for license information.
import React from 'react';
import {shallow} from 'enzyme';

import FileAttachment from './file_attachment.js';
import Preferences from 'mattermost-redux/constants/preferences';

jest.mock('react-native-doc-viewer', () => ({
    openDoc: jest.fn(),
}));

describe('FileAttachment', () => {
    const baseProps = {
        canDownloadFiles: true,
        file: {
            create_at: 1546893090093,
            delete_at: 0,
            extension: 'png',
            has_preview_image: true,
            height: 171,
            id: 'fileId',
            name: 'image.png',
            post_id: 'postId',
            size: 14894,
            update_at: 1546893090093,
            user_id: 'userId',
            width: 425,
            data: {
                mime_type: 'image/png',
            },
        },
        id: 'id',
        index: 0,
        theme: Preferences.THEMES.default,
    };

    test('should match snapshot', () => {
        const wrapper = shallow(
            <FileAttachment {...baseProps}/>,
        );

        expect(wrapper.getElement()).toMatchSnapshot();
    });
});
