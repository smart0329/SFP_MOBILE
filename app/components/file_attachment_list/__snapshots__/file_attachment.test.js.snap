// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FileAttachment should match snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "borderColor": "rgba(61,60,64,0.4)",
        "borderRadius": 5,
        "borderWidth": 1,
        "flex": 1,
        "flexDirection": "row",
        "marginTop": 10,
      },
    ]
  }
>
  <View
    style={
      Object {
        "marginHorizontal": 20,
        "marginVertical": 10,
      }
    }
  >
    <TouchableWithFeedbackIOS
      onPress={[Function]}
      type="opacity"
    >
      <FileAttachmentIcon
        file={
          Object {
            "mime_type": "image/png",
          }
        }
        iconHeight={48}
        iconWidth={36}
        onCaptureRef={[Function]}
        theme={
          Object {
            "awayIndicator": "#ffbc42",
            "buttonBg": "#166de0",
            "buttonColor": "#ffffff",
            "centerChannelBg": "#ffffff",
            "centerChannelColor": "#3d3c40",
            "codeTheme": "github",
            "dndIndicator": "#f74343",
            "errorTextColor": "#fd5960",
            "linkColor": "#2389d7",
            "mentionBg": "#ffffff",
            "mentionBj": "#ffffff",
            "mentionColor": "#145dbf",
            "mentionHighlightBg": "#ffe577",
            "mentionHighlightLink": "#166de0",
            "newMessageSeparator": "#ff8800",
            "onlineIndicator": "#06d6a0",
            "sidebarBg": "#145dbf",
            "sidebarHeaderBg": "#1153ab",
            "sidebarHeaderTextColor": "#ffffff",
            "sidebarText": "#ffffff",
            "sidebarTextActiveBorder": "#579eff",
            "sidebarTextActiveColor": "#ffffff",
            "sidebarTextHoverBg": "#4578bf",
            "sidebarUnreadText": "#ffffff",
            "type": "Mattermost",
          }
        }
        wrapperHeight={48}
        wrapperWidth={36}
      />
    </TouchableWithFeedbackIOS>
  </View>
</View>
`;
